File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int n = nums.size();
        int rot = k % n;
    reverse(nums.begin(), nums.end());
        reverse(nums.begin(), nums.begin() + rot);
        reverse(nums.begin() + rot, nums.end());
        

    }
};

File: 0189-rotate-array/0189-rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        k = k % len(nums)

        def reverse(l: int, r: int) -> None:
            while l <= r:
                nums[l], nums[r] = nums[r], nums[l]
                l += 1
                r -= 1

        # Step 1: Reverse the entire array
        reverse(0, len(nums) - 1)

        # Step 2: Reverse the first k elements (0 to k-1)
        reverse(0, k - 1)

        # Step 3: Reverse the remaining elements (k to len(nums)-1)
        reverse(k, len(nums) - 1)